tools/pannopi_report.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# @created: 15.05.2021
+# @author: Zilov Danil, GitHUB: zilov
+# @contact: zilov.d@gamail.com
+
+import argparse
+import glob
+import os
+
+def quast_report(quast_file):
+    quast_report = {}
+    with open(quast_file) as fh:
+        for line in fh:
+            line = line.strip().split("\t")
+            if line[0] == "# contigs":
+                contigs_n = line[1]
+                quast_report["Contigs_number"] = contigs_n
+            elif line[0] == "Total length (>= 0 bp)":
+                length = line[1]
+                quast_report["Total_length"] = length
+            elif line[0] == "N50":
+                n50 = line[1]
+                quast_report["N50"] = n50
+            elif line[0] == "# misassemblies":
+                misasemblies = line[1]
+                quast_report["Misassemblies"] = misasemblies
+            elif line[0] == "Genome fraction (%)":
+                genome_fraction = line[1]
+                quast_report["Genome_fraction"] = genome_fraction
+    return quast_report
+
+def prokka_report(prokka_file):
+    prokka_report = {}
+    with open(prokka_file) as fh:
+        for line in fh:
+            line = line.strip().split(": ")
+            if line[0] == "CDS":
+                cds = line[1]
+                prokka_report["CDS"] = cds
+            elif line[0] == "rRNA":
+                rrna = line[1]
+                prokka_report["rRNA"] = rrna
+            elif line[0] == "tRNA":
+                trna = line[1]
+                prokka_report["tRNA"] = trna
+    return prokka_report
+
+
+def fastani_report(fastani_file):
+    fastani_report = {}
+    with open(fastani_file) as fh:
+        for line in fh:
+            line = line.strip().split("\t")
+            fastani_report["FastANI"] = line[-3]
+            return fastani_report
+
+def busco_report(busco_dir):
+    busco_files = glob.glob(os.path.join(busco_dir, "*busco.txt"))
+    busco_report = {}
+    for file in busco_files:
+        with open(file) as fh:
+            for line in fh:
+                if line.startswith("# The lineage dataset is:"):
+                    dataset = "busco_" + line.strip().split()[5]
+                line = line.strip()
+                if line.startswith("C:"):
+                    buscos = line.strip().split("%")[0].split(":")[1]
```

Comment:
in each such case the good pattern to add example string it will save alotoftime

```
# string here that you are trying to parse
buscos = line.strip().split("%")[0].split(":")[1]
```